### PR TITLE
Add low_r signature grinding

### LINKED ISSUE #1387: Grind signatures to low-r value
It would be nice to have rust-lightning grind signatures to low-r value, possibly controlled by a feature gate.  This makes the tx weight more predictable.  Otherwise, there's a 50% chance for each signature to have an extra byte to represent the sign.

### PR DESCRIPTION
**Pull request by devrandom**
default on, can be turned off via a feature gate

Fixes #1387

### CODE REVIEW

File: lightning/Cargo.toml [hunk 1]
```diff
@@ -32,7 +32,10 @@ _bench_unstable = []
 no-std = ["hashbrown", "bitcoin/no-std", "core2/alloc"]
 std = ["bitcoin/std"]
 
-default = ["std"]
+# Generates low-r bitcoin signatures, which saves 1 byte in 50% of the cases
+grind_signatures = []
```
Comment (TheBlueMatt) [inline lightning/Cargo.toml:36]: Maybe lets flip the default? The grinding is quite fast, so in almost no cases should anyone ever care about it. Lets cheat, though, and have `_bench_unstable` rely on the non-grinding :)

**Review by TheBlueMatt — APPROVED** [review]

**Review by arik-so — APPROVED** [review]
awesome, thanks!

**Comment by valentinewallace** [comment]
It looks like this PR introduced a buncha warnings in CI, FWIW

**Comment by devrandom** [comment]
@valentinewallace I don't quite see how these changes would result in those warnings.  I assume you mean in channel.rs?

**Comment by valentinewallace** [comment]
@devrandom Yeah, I'm not sure how either, I just see that the warnings appear on this PR's merge. I see them in "Test on Rust stable with net-tokio" but they may be elsewhere too

**Comment by devrandom** [comment]
Oh, I know, that's in the `mod tests`.  One of the tests is only compiled in when grinding is off, because it hardcodes the check of a bunch of signatures.

**Comment by devrandom** [comment]
hmm... that test is a bit of a pain.  perhaps we can have it skip the signatures when comparing?

**Comment by devrandom** [comment]
@TheBlueMatt since you wrote `test_commitment!` - how do you feel about skipping the comparison of the signature?

**Pull request merged**